File: javax/src/test/java/me/panpf/javax/test/lang/NumberxTest.java
/*
 * Copyright (C) 2018 Peng fei Pan <panpfpanpf@outlook.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package me.panpf.javax.test.lang;

import me.panpf.javax.lang.Numberx;
import org.junit.Test;

import static org.junit.Assert.*;

public class NumberxTest {

    @Test
    public void testTo() {
        assertEquals(String.valueOf(Numberx.toByteOrDefault("5", (byte) 4)), String.valueOf(5));
        assertEquals(String.valueOf(Numberx.toByteOrDefault("g", (byte) 4)), String.valueOf(4));
        assertEquals(String.valueOf(Numberx.toByteOrZero("g")), String.valueOf(0));
        assertEquals(String.valueOf(Numberx.toByteOrDefault("", (byte) 4)), String.valueOf(4));
        assertEquals(String.valueOf(Numberx.toByteOrDefault(null, (byte) 4)), String.valueOf(4));

        assertEquals(String.valueOf(Numberx.toShortOrDefault("5", (short) 4)), String.valueOf(5));
        assertEquals(String.valueOf(Numberx.toShortOrDefault("g", (short) 4)), String.valueOf(4));
        assertEquals(String.valueOf(Numberx.toShortOrZero("g")), String.valueOf(0));
        assertEquals(String.valueOf(Numberx.toShortOrDefault("", (short) 4)), String.valueOf(4));
        assertEquals(String.valueOf(Numberx.toShortOrDefault(null, (short) 4)), String.valueOf(4));

        assertEquals(String.valueOf(Numberx.toIntOrDefault("5", 4)), String.valueOf(5));
        assertEquals(String.valueOf(Numberx.toIntOrDefault("g", 4)), String.valueOf(4));
        assertEquals(String.valueOf(Numberx.toIntOrZero("g")), String.valueOf(0));
        assertEquals(String.valueOf(Numberx.toIntOrDefault("", 4)), String.valueOf(4));
        assertEquals(String.valueOf(Numberx.toIntOrDefault(null, 4)), String.valueOf(4));

        assertEquals(String.valueOf(Numberx.toLongOrDefault("5", 4)), String.valueOf(5));
        assertEquals(String.valueOf(Numberx.toLongOrDefault("g", 4)), String.valueOf(4));
        assertEquals(String.valueOf(Numberx.toLongOrZero("g")), String.valueOf(0));
        assertEquals(String.valueOf(Numberx.toLongOrDefault("", 4)), String.valueOf(4));
        assertEquals(String.valueOf(Numberx.toLongOrDefault(null, 4)), String.valueOf(4));

        assertEquals(String.valueOf(Numberx.toFloatOrDefault("5.5", 4.5f)), String.valueOf(5.5f));
        assertEquals(String.valueOf(Numberx.toFloatOrDefault("g", 4.4f)), String.valueOf(4.4f));
        assertEquals(String.valueOf(Numberx.toFloatOrZero("g")), String.valueOf(0.0f));
        assertEquals(String.valueOf(Numberx.toFloatOrDefault("", 4.4f)), String.valueOf(4.4f));
        assertEquals(String.valueOf(Numberx.toFloatOrDefault(null, 4.4f)), String.valueOf(4.4f));

        assertEquals(String.valueOf(Numberx.toDoubleOrDefault("5.5", 4.4d)), String.valueOf(5.5d));
        assertEquals(String.valueOf(Numberx.toDoubleOrDefault("g", 4.4d)), String.valueOf(4.4d));
        assertEquals(String.valueOf(Numberx.toDoubleOrZero("g")), String.valueOf(0.0d));
        assertEquals(String.valueOf(Numberx.toDoubleOrDefault("", 4.4d)), String.valueOf(4.4d));
        assertEquals(String.valueOf(Numberx.toDoubleOrDefault(null, 4.4d)), String.valueOf(4.4d));
    }
}

File: javax-kt/src/test/java/me/panpf/javaxkt/test/lang/NumberxTest.kt
/*
 * Copyright (C) 2018 Peng fei Pan <panpfpanpf@outlook.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package me.panpf.javaxkt.test.lang

import me.panpf.javaxkt.lang.*
import org.junit.Assert.assertEquals
import org.junit.Test

class NumberxTest {

    @Test
    fun testTo() {
        assertEquals("5".toByteOrDefault(4.toByte()).toString(), 5.toString())
        assertEquals("g".toByteOrDefault(4.toByte()).toString(), 4.toString())
        assertEquals("g".toByteOrZero().toString(), 0.toString())
        assertEquals("".toByteOrDefault(4.toByte()).toString(), 4.toString())
        assertEquals(null.toByteOrDefault(4.toByte()).toString(), 4.toString())

        assertEquals("5".toShortOrDefault(4.toShort()).toString(), 5.toString())
        assertEquals("g".toShortOrDefault(4.toShort()).toString(), 4.toString())
        assertEquals("g".toShortOrZero().toString(), 0.toString())
        assertEquals("".toShortOrDefault(4.toShort()).toString(), 4.toString())
        assertEquals(null.toShortOrDefault(4.toShort()).toString(), 4.toString())

        assertEquals("5".toIntOrDefault(4).toString(), 5.toString())
        assertEquals("g".toIntOrDefault(4).toString(), 4.toString())
        assertEquals("g".toIntOrZero().toString(), 0.toString())
        assertEquals("".toIntOrDefault(4).toString(), 4.toString())
        assertEquals(null.toIntOrDefault(4).toString(), 4.toString())

        assertEquals("5".toLongOrDefault(4).toString(), 5.toString())
        assertEquals("g".toLongOrDefault(4).toString(), 4.toString())
        assertEquals("g".toLongOrZero().toString(), 0.toString())
        assertEquals("".toLongOrDefault(4).toString(), 4.toString())
        assertEquals(null.toLongOrDefault(4).toString(), 4.toString())

        assertEquals("5.5".toFloatOrDefault(4.5f).toString(), 5.5f.toString())
        assertEquals("g".toFloatOrDefault(4.4f).toString(), 4.4f.toString())
        assertEquals("g".toFloatOrZero().toString(), 0.0f.toString())
        assertEquals("".toFloatOrDefault(4.4f).toString(), 4.4f.toString())
        assertEquals(null.toFloatOrDefault(4.4f).toString(), 4.4f.toString())

        assertEquals("5.5".toDoubleOrDefault(4.4).toString(), 5.5.toString())
        assertEquals("g".toDoubleOrDefault(4.4).toString(), 4.4.toString())
        assertEquals("g".toDoubleOrZero().toString(), 0.0.toString())
        assertEquals("".toDoubleOrDefault(4.4).toString(), 4.4.toString())
        assertEquals(null.toDoubleOrDefault(4.4).toString(), 4.4.toString())
    }
}